docs_src/design/adr/security/0017-consul-security.md

```diff
@@ -0,0 +1,80 @@
+# Securing access to Consul
+
+## Status
+
+** proposed **
+
+## Context
+
+This ADR defines the motiviation and approach used to secure access
+to the Consul component in the EdgeX archticture.
+
+Consul provides several services for the EdgeX architecture:
+
+- Service registry
+- Service health monitoring
+- Mutable configuration data
+
+Use of the services provided by Consul is optional on a service-by-service basis.
+Use of the registry is controlled by the `--registry` flag provided to an EdgeX service.
+Use of mutable configuration data is controlled by the `-cp` flag provided to an EdgeX service.
+Where possible, a static `configuration.toml` file bundled with each service is used as a fallback mechanism.
+When Consul is enabled as a configuration provider,
+the `configuration.toml` is parsed into individual settings
+and seeded into the Consul key-value store on the first start of a service.
+Further accesses to configuration data are performed first against the Consul key-value store,
+and the static `configuration.toml` remains as the fallback mechanism.
+
+Since configuration data can affect the runtime behavior of services,
+compensating controls must be introduced in order to mitigate the risks introduced
+by moving configuration from a static file verified with a SHA-256 hash
```

Comment:
How are these file hashes being checked? Is this specific to docker deployment?

Reply:
Anything in the docker image is covered by the docker image's SHA256 hash.  I should mention digital signature for snaps.

Reply:
Decided instead not to dwell on the "why" for sake of brevity.